backend/attachRendererFiber.js

```diff
@@ -354,6 +370,14 @@ function attachRendererFiber(hook: Hook, rid: string, renderer: ReactRenderer):
         break;
     }
 
+    if (
+      props !== null &&
+      typeof fiber.elementType !== undefined &&
+      fiber.type !== fiber.elementType
+    ) {
+      props = resolveDefaultProps(fiber.type, props);
```

Comment:
I could've put this into a more specific place, like `Function` + `Class` + `ForwardRef` + `Memo` or whichever ones support it. But I found this easier to read. The only way you could have a false positive here is if somebody assigned `defaultProps` to e.g. a context consumer. I think that's unlikely enough that it would be fine for DevTools to mistakingly resolve them for a type that doesn't support them — to keep the code readable.

Reply:
This seems safe. I think any time `fiber.type !== fiber.elementType`, it's a lazy component.